src/sparsezoo/utils/data.py

```diff
@@ -62,6 +61,49 @@ def __iter__(self) -> Iterator[Union[numpy.ndarray, Dict[str, numpy.ndarray]]]:
         for item in self._data:
             yield item
 
+    def iter_batches(
+        self,
+        batch_size: int,
+        iterations: int,
+    ) -> Generator[List[numpy.ndarray], None, None]:
+        """
+        A method to iterate over data in batches
+
+        :param batch_size: Positive integer representing the size of each batch
+        :param iterations: Positive integer representing number of batches to yield
+
+        :pre: batch_size should be within the range [0, number_of_samples)
```

Comment:
`batch_size` can be any non-negative value since this implementation will keep looping over the dataset even if it starts repeating. this is good for cases where we have say one sample but want to measure across multiple batch sizes. no code change needed